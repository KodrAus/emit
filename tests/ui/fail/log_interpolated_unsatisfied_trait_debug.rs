#![feature(stmt_expr_attributes, proc_macro_hygiene)]

// Does not implement `Display`
struct Input;

fn main() {
    emit::info!("Text \"and\" {#[emit::as_debug] a: Input} and more");
}
