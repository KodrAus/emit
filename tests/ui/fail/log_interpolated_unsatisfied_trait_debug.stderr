error[E0277]: `Input` doesn't implement `Debug`
   --> fail/log_interpolated_unsatisfied_trait_debug.rs:7:32
    |
7   |     emit::info!("Text \"and\" {#[emit::as_debug] a: Input} and more");
    |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^ `Input` cannot be formatted using `{:?}`
    |
    = help: the trait `Debug` is not implemented for `Input`
    = note: add `#[derive(Debug)]` to `Input` or manually `impl Debug for Input`
    = help: the following other types implement trait `main::emit::emit_rt::capture::Capture<T>`:
              <(dyn Debug + 'static) as main::emit::emit_rt::capture::Capture<main::emit::emit_rt::capture::CaptureDebug>>
              <(dyn std::error::Error + 'static) as main::emit::emit_rt::capture::Capture<main::emit::emit_rt::capture::CaptureError>>
              <(dyn std::fmt::Display + 'static) as main::emit::emit_rt::capture::Capture<main::emit::emit_rt::capture::CaptureDisplay>>
              <(dyn sval::value::Value + 'static) as main::emit::emit_rt::capture::Capture<main::emit::emit_rt::capture::CaptureSval>>
              <str as main::emit::emit_rt::capture::Capture<T>>
    = note: required because of the requirements on the impl of `main::emit::emit_rt::capture::Capture<main::emit::emit_rt::capture::CaptureAnonDebug>` for `Input`
note: required by a bound in `__private_capture_anon_as_debug`
   --> $WORKSPACE/rt/src/capture.rs
    |
    |         Self: Capture<CaptureAnonDebug>,
    |               ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `__private_capture_anon_as_debug`
    = note: this error originates in the macro `emit::ct::__private_capture_anon_as_debug` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `Input` with `#[derive(Debug)]`
    |
4   | #[derive(Debug)]
    |
